### DISCUSSION TITLE
Controlling Stereolabs ZED mini camera from Bonsai

### DISCUSSION THREAD

**Discussion by rkast1** [bonsai-rx/bonsai#1111] [Q&A] [answered]
Hi all, I am fairly new to Bonsai and new to the discussion forum. I am attempting to record video through a ZED mini [stereocamera](https://store.stereolabs.com/products/zed-mini) using the VideoCaptureDevice Node. I can acquire the video and adjust the frame rate, but I would like to adjust other video settings including the exposure, gain and contrast. The ZED API seems quite flexible, but I am not sure how to go about controlling these VIDEO_SETTING features from within Bonsai. Here's a [link](https://www.stereolabs.com/docs/api/group__Video__group.html) to the API in case someone can make use of it to help me find a solution. Thanks in advance!

**Comment by bruno-f-cruz** [comment, marked as answer]
Currently, there is no official package for these cameras. However, it seems that the company has provided drivers for a generic directshow interface (this is what VideoCaptureDevices uses as a backend). You can try to adjust the camera settings there (although it probably won't be as flexible as their API). You can access the exposed settings under the "CaptureProperties" property or you can also click the little icon at the top to access the driver-specific GUI.

![image](https://user-images.githubusercontent.com/7049351/200123945-38e89cd9-cdba-4816-ad86-6208d7e45e43.png)

**Reply by rkast1** [reply]
Thanks, can you send me the link to the driver that you are referring to? This icon does not appear within my VideoCaptureDevice node:
![image](https://user-images.githubusercontent.com/48425934/200136808-ee2fd482-aa20-4fbb-9b1e-7bd8c06e0cda.png)

**Comment by bruno-f-cruz** [comment]
I should have been clearer. The driver is already installed and it is what
allows you to see the camera in the VideoCaptureDevice. Have you tried to
change the captureproperties ?

**Reply by rkast1** [reply]
I see. Thank you for clarifying. Yes, the only exposed setting within the CaptureProperties for this webcam is exposure. I have tried changing the value, but it doesn't seem to have much effect on the brightness of the frames, which is why I am reaching out about this issue in the first place. For a little more context, I am recording video under relatively low light conditions, and am hoping to make the images a bit brighter. Thanks again.
